overlord/snapshotstate/backend/backend.go

```diff
@@ -140,6 +144,17 @@ func Iter(ctx context.Context, f func(*Reader) error) error {
 			if !ok {
 				continue
 			}
+			// keep track of in-progress in a map as well
+			// to avoid races from the fact that we read only
+			// 100 dir entries at a time
+			if importsInProgress[setID] {
+				continue
+			}
+			if newImportTransaction(setID).InProgress() {
```

Comment:
I understand how it works, but this reads strangely, my preference would be to have a help that is not a method for this.

ImportInProgressFor(setID)